synchronizer/worker/largesegment/large_segment.go

```diff
@@ -0,0 +1,211 @@
+package largesegment
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/splitio/go-split-commons/v6/dtos"
+	"github.com/splitio/go-split-commons/v6/healthcheck/application"
+	"github.com/splitio/go-split-commons/v6/service"
+	"github.com/splitio/go-split-commons/v6/storage"
+	"github.com/splitio/go-split-commons/v6/synchronizer/worker/segment"
+	"github.com/splitio/go-toolkit/v5/backoff"
+	"github.com/splitio/go-toolkit/v5/datastructures/set"
+	"github.com/splitio/go-toolkit/v5/logging"
+	"golang.org/x/sync/semaphore"
+)
+
+const (
+	onDemandFetchBackoffBase       = int64(10)        // backoff base starting at 10 seconds
+	onDemandFetchBackoffMaxWait    = 60 * time.Second //  don't sleep for more than 1 minute
+	onDemandFetchBackoffMaxRetries = 10
+
+	maxConcurrency = 10
```

Comment:
concurrency should be less probably, remember that this is csv processing, not common segement